src/context.rs

```diff
@@ -1,88 +1,360 @@
+
+        impl<T, C> $crate::Pop<T> for $context_name<T, C> {
+            type Result = C;
+            fn pop(self) -> (T, Self::Result) {
+                (self.head, self.tail)
+            }
+        }
+
+        impl<C, T, U> $crate::Push<U> for $context_name<T, C> {
+            type Result = $context_name<U, Self>;
+            fn push(self, item: U) -> Self::Result {
+                $context_name{head: item, tail: self}
+            }
+        }
+
+        new_context_type!(impl extend_has $context_name, $empty_context_name, $($types),+);
+    };
+    (impl extend_has $context_name:ident, $empty_context_name:ident, $head:ty, $($tail:ty),+ ) => {
+        new_context_type!(
+            impl extend_has_helper
+            $context_name,
+            $empty_context_name,
+            $head,
+            $($tail),+
+        );
+        new_context_type!(impl extend_has $context_name, $empty_context_name, $($tail),+);
+    };
+    (impl extend_has $context_name:ident, $empty_context_name:ident, $head:ty) => {};
+    (impl extend_has_helper
+        $context_name:ident,
+        $empty_context_name:ident,
+        $type:ty,
+        $($types:ty),+
+        ) => {
+        $(
+            impl<C: $crate::Has<$type>> $crate::Has<$type> for $context_name<$types, C> {
+                fn set(&mut self, item: $type) {
+                    self.tail.set(item);
+                }
+
+                fn get(&self) -> &$type {
+                    self.tail.get()
+                }
+
+                fn get_mut(&mut self) -> &mut $type {
+                    self.tail.get_mut()
+                }
+            }
+
+            impl<C: $crate::Has<$types>> $crate::Has<$types> for $context_name<$type, C> {
+                fn set(&mut self, item: $types) {
+                    self.tail.set(item);
+                }
+
+                fn get(&self) -> &$types {
+                    self.tail.get()
+                }
+
+                fn get_mut(&mut self) -> &mut $types {
+                    self.tail.get_mut()
+                }
+            }
+
+            impl<C> $crate::Pop<$type> for $context_name<$types, C> where C: Pop<$type> {
+                type Result = $context_name<$types, C::Result>;
+                fn pop(self) -> ($type, Self::Result) {
+                    let (value, tail) = self.tail.pop();
+                    (value, $context_name{ head: self.head, tail})
+                }
+            }
+
+            impl<C> $crate::Pop<$types> for $context_name<$type, C> where C: Pop<$types> {
+                type Result = $context_name<$type, C::Result>;
+                fn pop(self) -> ($types, Self::Result) {
+                    let (value, tail) = self.tail.pop();
+                    (value, $context_name{ head: self.head, tail})
+                }
+            }
+        )+
+    };
+}
+
+/// Create a default context type to export.
+new_context_type!(Context, EmpContext, XSpanIdString, Option<AuthData>, Option<Authorization>);
```

Comment:
Oh, I guess this creates the empty context that people are supposed to use? Not sure it's worth saving the extra 2 characters for `EmptyContext` (or `BaseContext`?)

Reply:
Yeah, that was the plan but as I said [https://github.com/Metaswitch/swagger-rs/pull/29#discussion_r181004006](here) I don't think it's needed anymore.